torch/testing/_internal/opinfo_helper.py

```diff
@@ -0,0 +1,110 @@
+from collections import defaultdict
+import warnings
+from functools import partial
+
+import torch
+from torch.testing._internal.common_cuda import (TEST_CUDA)
+from torch.testing._internal.common_methods_invocations import (op_db, _DYNAMIC_DTYPES)
+from torch.testing import (all_types_and_complex_and,
+                           all_types_and_complex,
+                           # all_types_and_half, # not exported by testing at the moment
+                           all_types,
+                           complex_types,
+                           floating_and_complex_types,
+                           # floating_types_and_half, # not exported by testing at the moment
+                           floating_types,
+                           integral_types,
+                           )
+
+DTYPE_DISPATCH = (
+    floating_types,
+    floating_and_complex_types,
+    integral_types,
+    all_types,
+    complex_types,
+    all_types_and_complex,
+)
+
+# Better way to acquire devices?
+DEVICES = ['cpu'] + (['cuda'] if TEST_CUDA else [])
+
+if not TEST_CUDA:
```

Comment:
We should switch this check to be if the CUDA device type was returned by the torch.testing function suggested above and, if not, warn that the ops supported datatypes on a CUDA device couldn't be determined because no CUDA device was available

Reply:
Also, is this the best place for this warning? If we think of the future of this tool facilitating a variety of user-facing functions, maybe it should have a warn once behavior when a function that would normally return info about CUDA is called?